test-network-function/suite_test.go

```diff
@@ -243,9 +244,9 @@ func generateNodes() map[string]interface{} {
 		csiDriverInfo    = "csiDriver"
 	)
 	nodes := map[string]interface{}{}
-	nodes[nodeSummaryField] = diagnostic.GetNodeSummary()
-	nodes[cniPluginsField] = diagnostic.GetCniPlugins()
-	nodes[nodesHwInfo] = diagnostic.GetNodesHwInfo()
-	nodes[csiDriverInfo] = diagnostic.GetCsiDriverInfo()
+	nodes[nodeSummaryField] = make(map[string]interface{})
+	nodes[cniPluginsField] = make([]diagnostic.CniPlugin, 0)
+	nodes[nodesHwInfo] = diagnostic.NodesHwInfo{}
+	nodes[csiDriverInfo] = diagnostic.NewCsiDriverInfo()
```

Comment:
I think this is not going to work. Those fields in the "nodes" map are used to report the k8s cluster information in the claim.json file, and with this we are just creating empty structs. Thus, we're always getting null information from the test suite.

Something similar is happening in test-network-function/diagnostic/suite.go, where the global variables are replaced by local ones in some cases, so the recorded information is never propagated outside the test itself.

Reply:
Looking at this, I agree.  I'll mark this work-in-progress.

The main thing that I don't like about this section of code is all of the global variables populated by what is pretty much an `init` function.